From 41836d5f080db62106ec1e2975dfb52eb7cb0910 Mon Sep 17 00:00:00 2001
From: Szehon Ho <szehon@cloudera.com>
Date: Thu, 1 Oct 2015 11:33:39 -0700
Subject: [PATCH 0330/1164] CDH-29870: more HIVE-11982 : Some test cases for
 union all fail with recent changes (Yongzhi Chen
 via Szehon)

---
 .../hive/ql/optimizer/ColumnPrunerProcCtx.java     |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ql/src/java/org/apache/hadoop/hive/ql/optimizer/ColumnPrunerProcCtx.java b/ql/src/java/org/apache/hadoop/hive/ql/optimizer/ColumnPrunerProcCtx.java
index 8bcb464..2207cfb 100644
--- a/ql/src/java/org/apache/hadoop/hive/ql/optimizer/ColumnPrunerProcCtx.java
+++ b/ql/src/java/org/apache/hadoop/hive/ql/optimizer/ColumnPrunerProcCtx.java
@@ -270,7 +270,7 @@ public void handleFilterUnionChildren(Operator<? extends OperatorDesc> curOp)
 
     for (Operator<? extends OperatorDesc> child : curOp.getChildOperators()) {
       if (child instanceof UnionOperator) {
-        prunList = prunedColLists.get(child);
+        prunList = genColLists(curOp, child);
         if (prunList == null || prunList.size() == 0 || parentPrunList.size() == prunList.size()) {
           continue;
         }
-- 
1.7.9.5

